tmt/steps/report/reportportal.py

```diff
@@ -143,8 +153,82 @@ def go(self) -> None:
+                uuid = re_result.group()
+
+            if not isinstance(uuid, str):
+                self.debug(f"Could not update metadata in {message}")
+                return
+
+            # Get the launch_id from MQ uuid
+            with tmt.utils.retry_session() as session:
+                if self.get("ca-cert-file"):
+                    session.verify = self.get("ca-cert-file")
+                url = f"{server}/api/v1/{project}/launch/uuid/{uuid}"
+                self.debug(f"Send the report to '{url}'.")
+                response = session.get(
+                    url,
+                    headers={
+                        "Authorization": "bearer " + token,
+                        "accept": "*/*",
+                        },
+                    )
+                # Handle the response
+                try:
+                    message = tmt.utils.yaml_to_dict(response.text).get("message")
+                except (tmt.utils.GeneralError, KeyError):
+                    message = response.text
+                if not response.ok:
+                    raise tmt.utils.ReportError(
+                        "Received non-ok status code from ReportPortal, "
+                        f"response text is: {message}")
+                else:
+                    self.debug(f"Response code from the server: {response.status_code}")
+                    self.debug(f"Message from the server: {message}")
+                    self.info("report", "Successfully uploaded.", "yellow")
+
+                launch_id = json.loads(response.text)["id"]
+                if launch_id:
+                    with tmt.utils.retry_session() as session:
+                        if self.get("ca-cert-file"):
+                            session.verify = self.get("ca-cert-file")
+                        url = f"{server}/api/v1/{project}/launch/{launch_id}/update"
+                        self.debug(f"Send the report to '{url}'.")
+                        response = session.put(
+                            url,
+                            headers={
+                                "Content-Type": "application/json",
+                                "Authorization": "bearer " + token,
+                                "accept": "*/*",
+                                },
+                            data=json.dumps({"attributes": meta_data}),
+                            )
+
+                        # Handle the response
+                        try:
+                            message = tmt.utils.yaml_to_dict(response.text).get("message")
+                        except (tmt.utils.GeneralError, KeyError):
+                            message = response.text
+                        if not response.ok:
+                            raise tmt.utils.ReportError(
+                                "Received non-ok status code from ReportPortal, "
+                                f"response text is: {message}")
+                        else:
```

Comment:
Same as above, `else` can be simplified thanks to the positive branch ending with `raise`.

Reply:
Ack